### ISSUE TITLE
TLS 1.3 client: Parsing of the early data indication extension

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/mbedtls#6328]
This issue tracks the up-streaming on client side of the parsing of the early data indication extension

Source: https://github.com/hannestschofenig/mbedtls/tree/tls13-prototype

PLEASE READ THOROUGHLY section 4.2.10 of the specification

- [ ] Add parsing of the early data indication extension in the EncryptedExtension message 

Testing:
test (ssl-opt.sh test) with an OpenSSL/GnuTLS server that an EncryptedExtension message containing an early data indication extension is successfully parsed. The resumption handshake eventually should fail as long as the client does not write the End of Early Data message in response (another issue dedicated to that).

Depends on #6326
